core/src/main/java/org/apache/accumulo/testing/core/randomwalk/multitable/CopyTool.java

```diff
@@ -64,70 +56,21 @@ public int run(String[] args) throws Exception {
       return 1;
     }
 
-    ClientConfiguration clientConf = ClientConfiguration.create().withInstance(args[3]).withZkHosts(args[4]);
-
+    ClientInfo info = Accumulo.newClient().usingProperties(args[0]).info();
     job.setInputFormatClass(AccumuloInputFormat.class);
-    AccumuloInputFormat.setInputTableName(job, args[2]);
+    AccumuloInputFormat.setClientInfo(job, info);
+    AccumuloInputFormat.setInputTableName(job, args[1]);
     AccumuloInputFormat.setScanAuthorizations(job, Authorizations.EMPTY);
-    AccumuloInputFormat.setZooKeeperInstance(job, clientConf);
-
-    final String principal;
-    final AuthenticationToken token;
-    if (Boolean.parseBoolean(clientConf.get(ClientProperty.INSTANCE_RPC_SASL_ENABLED))) {
-      // Use the Kerberos creds to request a DelegationToken for MapReduce
-      // to use
-      // We could use the specified keytab (args[1]), but we're already
-      // logged in and don't need to, so we can just use the current user
-      KerberosToken kt = new KerberosToken();
-      try {
-        UserGroupInformation user = UserGroupInformation.getCurrentUser();
-        if (!user.hasKerberosCredentials()) {
-          throw new IllegalStateException("Expected current user to have Kerberos credentials");
-        }
-
-        // Get the principal via UGI
-        principal = user.getUserName();
-
-        // Connector w/ the Kerberos creds
-        ZooKeeperInstance inst = new ZooKeeperInstance(clientConf);
-        Connector conn = inst.getConnector(principal, kt);
-
-        // Do the explicit check to see if the user has the permission
-        // to get a delegation token
-        if (!conn.securityOperations().hasSystemPermission(conn.whoami(), SystemPermission.OBTAIN_DELEGATION_TOKEN)) {
-          log.error(principal + " doesn't have the " + SystemPermission.OBTAIN_DELEGATION_TOKEN.name()
-              + " SystemPermission neccesary to obtain a delegation token. MapReduce tasks cannot automatically use the client's"
-              + " credentials on remote servers. Delegation tokens provide a means to run MapReduce without distributing the user's credentials.");
-          throw new IllegalStateException(conn.whoami() + " does not have permission to obtain a delegation token");
-        }
-
-        // Fetch a delegation token from Accumulo
-        token = conn.securityOperations().getDelegationToken(new DelegationTokenConfig());
```

Comment:
So this isn't needed anymore?

Reply:
The delegation token should be created outside of this code and set in the client properties that are passed into the MapReduce code.

Reply:
> The delegation token should be created outside of this code and set in the client properties that are passed into the MapReduce code.

That is a more elegant way to use delegation tokens.  Then there is no need to write special code for each map reduce job.